eslint/babel-eslint-parser/src/configuration.js

```diff
@@ -41,6 +26,7 @@ export function normalizeBabelParseConfig(options) {
     caller: {
       name: "@babel/eslint-parser",
     },
+    ...options.babelOptions,
```

Comment:
I would prefer to deeply merge the caller and the parser options, similarly to how we do for `caller` in `babel-loader` ([source](https://github.com/babel/babel-loader/blob/adac4592274e151ebf0313a33da553a19a8bf39d/src/injectCaller.js#L6)):

```js
    ...options.babelOptions,
    parserOpts: {
      allowImportExportEverywhere: options.allowImportExportEverywhere,
      allowReturnOutsideFunction: true,
      allowSuperOutsideMethod: true,
    ...options.babelOptions.parserOpts,
      plugins: ["estree", ...options.babelOptions.parserOpts?.plugins ?? []],
      ranges: true,
      tokens: true,
    },
    caller: {
      name: "@babel/eslint-parser",
    ...options.babelOptions.caller,
    },
```

By doing so, we make sure to keep out defaults (especially for the parser options) even if someone overwrites some of them.

Reply:
Makes sense!